wear/src/main/kotlin/au/com/shiftyjelly/pocketcasts/wear/MainActivity.kt

```diff
@@ -32,13 +39,40 @@ class MainActivity : ComponentActivity() {
 
     @Inject lateinit var theme: Theme
 
+    private val viewModel: WearMainActivityViewModel by viewModels()
+
+    private val dataClient by lazy { Wearable.getDataClient(this) }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContent {
-            // TODO add lines for radioactive theme
+            DisposableEffect(lifecycle) {
+                monitorDataLayerForChanges()
+            }
+
             WearApp(theme.activeTheme)
         }
     }
+
+    private fun DisposableEffectScope.monitorDataLayerForChanges(): DisposableEffectResult {
+
+        // immediately check for any changes on launch
+        viewModel.checkLatestSyncData()
+
+        // listen for changes after launch
+        val listener = LifecycleEventObserver { _, event ->
+            when (event) {
+                Lifecycle.Event.ON_START -> dataClient.addListener(viewModel.phoneSyncDataListener)
+                Lifecycle.Event.ON_STOP -> dataClient.removeListener(viewModel.phoneSyncDataListener)
+                else -> { /* do nothing */
+                }
+            }
+        }
+        lifecycle.addObserver(listener)
+        return onDispose {
+            lifecycle.removeObserver(listener)
+        }
```

Comment:
Instead of using a listener, we could use [a service](https://developers.google.com/android/reference/com/google/android/gms/wearable/WearableListenerService) to always listen for changes even when the Pocket Casts wear app is in the background. I don't think we need to listen and respond to these events until the app comes to the foreground, so I went with the listener approach. I do think either approach could work though, so let me know if you have other thoughts.